docs/provisioner/provisioning-api.md

```diff
@@ -85,25 +85,49 @@ The mutation returns OperationID allowing to retrieve the operation status.
 - Kubernetes cluster settings
   - Version
 
-The mutation returns OperationID allowing to retrieve the operation status.
+The mutation returns the operation ID which allows you to retrieve the operation status.
+
+### Upgrade Shoot cluster mutation
+
+***upgradeShoot*** mutation upgrades the Gardener Shoot cluster configuration, triggering Shoot reconciliation.
+
+>**NOTICE:** This operation is dangerous and could potentially damage your Runtime. Use it wisely and with reasonable configuration values. **There is no rollback functionality implemented**, so you would have to get your hands dirty to fix potential failures.
+The object passed to the mutation contains these configurable `GardenerConfig` values:
+
+| Field                                 |                                       Configurable                                       | Note         |
+| ------------------------------------- | :--------------------------------------------------------------------------------------: | ------------ |
+| **kubernetesVersion**                   |                                    :heavy_check_mark:                                    | upgrade only |
+| **purpose**                             |                                    :heavy_check_mark:                                    |              |
+| **machineType**                         |                                    :heavy_check_mark:                                    |              |
+| **volumeSizeGB**                        |                                    :heavy_check_mark:                                    |              |
+| **diskType**                            |                                    :heavy_check_mark:                                    |              |
+| **autoScalerMin**                       |                                    :heavy_check_mark:                                    | min. 1       |
+| **autoScalerMax**                       |                                    :heavy_check_mark:                                    |              |
+| **maxSurge**                            |                                    :heavy_check_mark:                                    |              |
+| **maxUnavailable**                      |                                    :heavy_check_mark:                                    |              |
+| **enableKubernetesVersionAutoUpdate**   |                                    :heavy_check_mark:                                    |              |
+| **enableMachineImageVersionAutoUpdate** |                                    :heavy_check_mark:                                    |              |
+| **providerSpecificConfig**              | Azure :heavy_check_mark: <br/> AWS :heavy_multiplication_x: <br/> GCP :heavy_check_mark: | `zones` only |
```

Comment:
I just came out with the idea that we should also add fields that are present in ProvisionClusterInput mutation, but are not supported by us - I had it in original version but it disappeared on migration to this PR (so it's quite ease to recover, but some values may require update) - because this table could be just condensed to:

```txt
We support all provided fields but in case of `kubernetesVersion` only upgrade and in case of `providerSpecificConfig` only `zones` for GCP and Azure.
```

What do you think?

Reply:
As discussed, I think we could add it as a follow-up task. Thanks for the comment!